test/lib/BedrockTester.cpp

```diff
@@ -583,4 +583,36 @@ int BedrockTester::waitForPort(int port) {
     return 1;
 }
 
+STable BedrockTester::getStatus(bool control) {
+    // FYI: sometimes the status command doesn't return with every key/value expected
+    SData request("Status");
+    return SParseJSONObject(executeWaitVerifyContent(request, "200", control));
+}
+
+string BedrockTester::getStatusTerm(string term, bool control) {
+    STable responseTable;
+    int i = 0;
+
+    // FYI: sometimes the status command doesn't return with every key/value expected
+    // loop up to 1000 times until you find it
+    while (!responseTable.count(term) && i<1000) {
+        responseTable = getStatus(control);
+        i++;
+    }
+    SASSERT(i<1000);
+    return responseTable[ term ];
+}
+
+bool BedrockTester::waitForCommit(int minCommitCount, int retries, bool control){
+    int commitCount = 0;
+    int i = 0;
+
+    // check commitCount up to "retries" times with a 1s sleep between calls
+    while (commitCount < minCommitCount && i < retries) {
+        sleep(1);
```

Comment:
the way this is written this call will always take a minimum of 1s. you can init the `commitCount` to the status output, which would potentially skip this 1s sleep if it's not needed on the first pass.

Reply:
Good point. Addressed.